src/test/service-worker_test.ts

```diff
@@ -185,6 +185,34 @@ suite('service-worker', () => {
           });
     });
 
+    test('basePath should prefix resources', () => {
+      return serviceWorker
+          .generateServiceWorker({
+            project: defaultProject,
+            buildRoot: testBuildRoot,
+            basePath: '/my/base/path'
+          })
+          .then((swFile: Buffer) => {
+            const fileContents = swFile.toString();
+            assert.include(fileContents, '"/my/base/path/index.html"');
+          });
+    });
+
+    test('basePath prefixes should not have double delimiters', () => {
```

Comment:
I think all these tests could be redone as async since generateServiceWorker is async
```typescript
test('...', async() => {
  const swFile = await serviceWorker.generateServiceWorker(...);
  const fileCOntents = swFile.toString();
  assert.include(...);
  ...
});
```